fe/src/main/java/org/apache/doris/catalog/PartitionInfo.java

```diff
@@ -127,7 +127,7 @@ public void write(DataOutput out) throws IOException {
         out.writeInt(idToDataProperty.size());
         for (Map.Entry<Long, DataProperty> entry : idToDataProperty.entrySet()) {
             out.writeLong(entry.getKey());
-            if (entry.getValue() == DataProperty.DEFAULT_HDD_DATA_PROPERTY) {
+            if (entry.getValue() == DataProperty.DEFAULT_DATA_PROPERTY) {
```

Comment:
Here will be a problem.
For now, `DataProperty.DEFAULT_DATA_PROPERTY` is configuration, so here, it is all possible to write `true` or `false`. And when reading, `true` may point to HDD, may point to SSD, depends on the config.

Reply:
I am confused why we should add a judge whether the DataProperty is default, can we just write `entry.getValue().write(out);`  and read as `DataProperty.read(in)`. I think the idToDataProperty's values can't be null?

Reply:
If we need to compatible with old version, I can chnage it to `entry.getValue() == new DataProperty(TStorageMedium.HDD)`

Reply:
You are righ, it is a little bit confused..
You can change it as you said.

Reply:
done